addons/account/controllers/portal.py

```diff
@@ -130,3 +143,17 @@ def details_form_validate(self, data, validate_main_address=True):
                 error['company_name'] = 'error'
                 error_message.append(_('Changing your company name is not allowed once invoices have been issued for your account. Please contact us directly for this operation.'))
         return error, error_message
+
+    def _invoice_get_searchbar_inputs(self):
+        values = {
+            'name': {'input': 'name', 'label': _('Search in #'), 'order': 1},
+        }
+        return dict(sorted(values.items(), key=lambda item: item[1]["order"]))
+
+    def _invoice_get_search_domain(self, search_in, search):
+        search_domain = []
+        if search_in in ('name', 'all'):
+            search_domain.append([('name', 'ilike', search)])
+        if search_in in ('ref', 'all'):
+            search_domain.append([('ref', 'ilike', search)])
+        return OR(search_domain)
```

Comment:
Seems could be simplified in most addons as you always have a single search_in -> single leaf domain.

Reply:
In case there is the 'all' search_in, and multiple other choices, the OR is required, I removed the function everywhere there was only one leaf